FrameworkSystem/scripts/dirac-populate-component-db.py

```diff
@@ -0,0 +1,138 @@
+                        '  debug:  Debug mode on/off'] ) )
+
+Script.parseCommandLine()
+args = Script.getPositionalArgs()
+
+if len( args ) > 0 and args[0] == 'debug':
+  debug = True
+else:
+  debug = False
+
+componentType = ''
+
+# Retrieve information from all the hosts
+client = SystemAdministratorIntegrator()
+resultAll = client.getOverallStatus()
+resultHosts = client.getHostInfo()
+resultInfo = client.getInfo()
+resultAllDB = client.getDatabases()
+resultAvailableDB = client.getAvailableDatabases()
+
+if not resultAll[ 'OK' ]:
+  print resultAll[ 'Message' ]
+elif not resultHosts[ 'OK' ]:
+  print resultHosts[ 'Message' ]
+elif not resultAllDB[ 'OK' ]:
+  print resultAllDB[ 'Message' ]
+elif not resultAvailableDB[ 'OK' ]:
+  print resultAvailableDB[ 'Message' ]
+else:
+  records = []
+  for host in resultAll[ 'Value' ]:
+    result = resultAll[ 'Value' ][ host ]
+    hostResult = resultHosts[ 'Value' ][ host ]
+    allDBResult = resultAllDB[ 'Value' ][ host ]
+    availableDBResult = resultAvailableDB[ 'Value' ][ host ]
+
+    if not result[ 'OK' ]:
+      if debug:
+        print 'Host %s: %s' % ( host, result[ 'Message' ] )
+      continue
+    elif not hostResult[ 'OK' ]:
+      if debug:
+        print 'Host %s: %s' % ( host, hostResult[ 'Message' ] )
+      continue
+    elif not allDBResult[ 'OK' ]:
+      if debug:
+        print 'Host %s: %s' % ( host, allDBResult[ 'Message' ] )
+      continue
+    elif not availableDBResult[ 'OK' ]:
+      if debug:
+        print 'Host %s: %s' % ( host, availableDBResult[ 'Message' ] )
+      continue
+
+    cpu = hostResult[ 'Value' ][ 'CPUModel' ].lstrip().rstrip()
+    rDict = result[ 'Value' ]
+    # Components other than databases
+    for compType in rDict:
+      if componentType and componentType != compType:
+        continue
+      for system in rDict[ compType ]:
+        components = rDict[ compType ][ system ].keys()
+        components.sort()
+        for component in components:
+          record = { 'Installation': {}, 'Component': {}, 'Host': {} }
+          if rDict[ compType ][ system ][ component ][ 'Installed' ]:
+            runitStatus = \
+                str( rDict[ compType ][ system ][ component ][ 'RunitStatus' ] )
+            if runitStatus != 'Unknown':
+              module = \
+                    str( rDict[ compType ][ system ][ component ][ 'Module' ] )
+              record[ 'Component' ][ 'System' ] = system
+              record[ 'Component' ][ 'Module' ] = module
+              record[ 'Component' ][ 'Type' ] = compType.lower()[ : -1 ]
+              record[ 'Host' ][ 'HostName' ] = host
+              record[ 'Host' ][ 'CPU' ] = cpu
+              record[ 'Installation' ][ 'Instance' ] = component
+              record[ 'Installation' ][ 'InstallationTime' ] = datetime.now()
+              records.append( record )
+
+    # Databases
+    csClient = CSAPI()
+    cfg = csClient.getCurrentCFG()[ 'Value' ]
+    setup = cfg.getOption( 'DIRAC/Setup' )
```

Comment:
In this case, using getSetup would only result in adding an extra import to the code, so I'm not sure about whether or not using it (unless it's mandatory)